corehq/apps/reports/filters/api.py

```diff
@@ -151,6 +153,86 @@ def get_groups(self, query, start, size):
         return [self.utils.reporting_group_tuple(g) for g in groups.run().hits]
 
 
+class LocationRestrictedEmwfOptionsMixin(object):
```

Comment:
[not blocking] I believe all of this can go in the base `EmwfOptionsView`, since if `can_access_all_locations`, the behavior is unchanged, and if not, the modified behavior would be correct.  I'm going to be using this modified EMWF soon to make the Submit History Report location safe, so I can also just merge the classes then if you prefer.